variants/STM32F7xx/F745Z(E-G)T_F746Z(E-G)(T-Y)_F750Z8T_F756ZG(T-Y)/variant_ETHERCAT_DUINO.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2020-2021, STMicroelectronics
+ * All rights reserved.
+ *
+ * This software component is licensed by ST under BSD 3-Clause license,
+ * the "License"; You may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *                        opensource.org/licenses/BSD-3-Clause
+ *
+ *******************************************************************************
+ */
+#if defined(ARDUINO_ETHERCAT_DUINO)
+#include "pins_arduino.h"
+
+// Digital PinName array
+const PinName digitalPin[] = {
+      PC_11  // D0    
+    , PC_10  // D1    
+    , PB_14  // D2    
+    , PD_12  // D3    
+    , PD_13  // D4    
+    , PD_14  // D5    
+    , PD_15  // D6    
+    , PC_6   // D7  
+    , PC_7   // D8  
+    , PC_8   // D9  
+    , PC_9   // D10 
+    , PA_8   // D11 
+    , PA_9   // D12 
+    , PA_15  // D13   
+    , PD_5   // D14 
+    , PD_6   // D15 
+    , PB_10  // D16   
+    , PB_11  // D17   
+    , PE_8   // D18 
+    , PE_7   // D19 
+    , PF_15  // D20   
+    , PF_14  // D21   
+    , PG_6   // D22 
+    , PG_7   // D23 
+    , PG_4   // D24 
+    , PG_5   // D25 
+    , PG_2   // D26 
+    , PG_3   // D27 
+    , PE_11  // D28   
+    , PE_15  // D29   
+    , PE_9   // D30 
+    , PE_10  // D31   
+    , PG_0   // D32 
+    , PG_1   // D33 
+    , PF_13  // D34   
+    , PF_12  // D35   
+    , PF_11  // D36   
+    , PG_9   // D37 
+    , PG_10  // D38   
+    , PG_11  // D39   
+    , PG_12  // D40   
+    , PG_13  // D41   
+    , PG_14  // D42   
+    , PF_10  // D43   
+    , PF_4   // D44 
+    , PF_3   // D45 
+    , PC_14  // D46   
+    , PC_15  // D47   
+    , PC_13  // D48   
+    , PE_6   // D49 
+    , PE_5   // D50 
+    , PE_4   // D51 
+    , PE_3   // D52 
+    , PE_2   // D53 
+};
+
+// Analog (Ax) pin number array
+const uint32_t analogInputPin[] = {
+      32 // A0 
+    , 33 // A1 
+    , 34 // A2 
+    , 35 // A3 
+    , 0  // A4 
+    , 1  // A5 
+    , 2  // A6 
+    , 3  // A7 
+    , 6  // A8 
+    , 7  // A9 
+    , 36 // A10
+    , 37 // A11
```

Comment:
You defined only 12 Analog pins while in the variant.h 
` #define NUM_ANALOG_INPUTS       24`

The check variant should have result failed.

Reply:
With 
```
#define NUM_DIGITAL_PINS             114 OR 54
#define NUM_ANALOG_INPUTS       24
```
CheckVariant compiles successfully. 
With `#define NUM_ANALOG_INPUTS       12` it fails. That is why I left both at the default for that chip, 114 and 24. Any idea why it fails?

Reply:
> CheckVariant compiles successfully.

The goal is not only to compile but to execute it to have the verdict: Passed  or Failed.

Reply:
Okay, that's maybe a stupid thing of me to not do haha, sorry for that.

If I compile with `NUM_ANALOG_INPUTS` defined as 12 I get the following error:
```
 52 | #define PC2                     PIN_A12
      |                                 ^~~~~~~
c:\Users\piete\Downloads\CheckVariant\digitalPinsTest.ino:2:27: note: in expansion of macro 'PC2'
    2 | #define PNUM(Port, Pin)   P ## Port ## Pin
      |                           ^
c:\Users\piete\Downloads\CheckVariant\digitalPinsTest.ino:4:105: note: in expansion of macro 'PNUM'
    4 | #define DECLTESTPNUM(Port, Pin)  checkPinNumber(XSTR(PNAME(Port, Pin)), PNAME(Port, Pin), "P"#Port#Pin, PNUM(Port, Pin))
      |                                                                                                         ^~~~
c:\Users\piete\Downloads\CheckVariant\digitalPinsTest.ino:147:8: note: in expansion of macro 'DECLTESTPNUM'
  147 |   if (!DECLTESTPNUM(C, 2)) testPassed = false;
```

When going to the definition of PIN_A12 it leads me to pins_arduino_analog.h which contains the following:
```
#if NUM_ANALOG_INPUTS > 12
  #define PIN_A12      (PIN_A11 + 1)
#endif
#if NUM_ANALOG_INPUTS > 13
  #define PIN_A13      (PIN_A12 + 1)
#endif
#if NUM_ANALOG_INPUTS > 14
  #define PIN_A14      (PIN_A13 + 1)
#endif
```

So it's fairly clear why the compiler fails. But I don't really see how I can solve this problem. I can change the `NUM_ANALOG_INPUTS` to 24 and declare all the pins in the analog input array but then the pin names would not correspond to the names printed on the silkscreen of our board. Do you have an idea how to solve this?

Reply:
You have to remove all `PIN_Ax `> 11 in the variant.h because they are defined based on the `NUM_ANALOG_INPUTS`
Your variant define which pins have Analog capacity. So if you provide only 12 analog then only PIN_A0 to PIN_A11 have to be define.

Reply:
It was kind of a struggle but I have managed to make it all work: 
```
12:7:38.734 -> #####
12:7:38.737 -> NUM_DIGITAL_PINS = 62
12:7:38.737 -> NUM_ANALOG_INPUTS = 12
12:7:38.737 -> #####
12:7:38.737 -> Checking digital pins...
12:7:38.738 -> End check digital pins
12:7:38.738 -> #####
12:7:38.738 -> Checking analog pins definition...
12:7:38.749 -> End check analog pins
12:7:38.749 -> #####
12:7:38.749 -> Checking IP instances (I2C, UART, SPI)...
12:7:38.768 -> End check IP instances (I2C, UART, SPI)
12:7:38.787 -> 
12:7:38.787 -> ########################################
12:7:38.787 -> #### Test PASSED
12:7:38.787 -> ########################################
12:7:38.787 -> 
```

Reply:
Hi @pieterconradie 
About analog pins, in fact you defined 0 analog pins as you have removed al the PIN_Ax. Moreover the `analogInputPin[] `defin pins with no analog capabilities (ex: `D32 `which is `PG0`).

Reply:
Hi @pieterjanbuntinx,
I think this comment is supposed to be for you.
Kind regards,
Pieter

Reply:
Thank you @pieterconradie  you are right. My mistake 😖

Reply:
@fpistm No worries :) Keep up the great work!